src/cobra/core/gene.py

```diff
@@ -299,3 +294,289 @@ def _repr_html_(self):
+        if knockouts is None:
+            knockouts = set()
+        if hasattr(self, "body"):
+            return self.__eval_gpr(self.body, knockouts=knockouts)
+        else:
+            return True
+
+    def __ast2str(
+        self,
+        expr: Union[Expression, BoolOp, Name, list],
+        level: int = 0,
+        names: dict = None,
+    ) -> str:
+        """convert compiled ast to gene_reaction_rule str
+
+        Parameters
+        ----------
+        expr : AST or GPR or list or Name or BoolOp
+            string for a gene reaction rule, e.g "a and b"
+        level : int
+            internal use only
+        names : dict
+            Dict where each element id a gene identifier and the value is the
+            gene name. Use this to get a rule str which uses names instead. This
+            should be done for display purposes only. All gene_reaction_rule
+            strings which are computed with should use the id.
+
+        Returns
+        ------
+        string
+            The gene reaction rule
+        """
+        if isinstance(expr, Expression) | isinstance(expr, GPR):
+            return self.__ast2str(expr.body, 0, names) if hasattr(expr, "body") else ""
+        elif isinstance(expr, Name):
+            return names.get(expr.id, expr.id) if names else expr.id
+        elif isinstance(expr, BoolOp):
+            op = expr.op
+            if isinstance(op, Or):
+                str_exp = " or ".join(
+                    self.__ast2str(i, level + 1, names) for i in expr.values
+                )
+            elif isinstance(op, And):
+                str_exp = " and ".join(
+                    self.__ast2str(i, level + 1, names) for i in expr.values
+                )
+            else:
+                raise TypeError("unsupported operation " + op.__class__.__name)
+            return "(" + str_exp + ")" if level else str_exp
+        elif expr is None or (isinstance(expr, list) and len(expr) == 0):
+            return ""
+        else:
+            raise TypeError("unsupported operation  " + repr(expr))
+        pass
+
+    def to_string(self, names: dict = None) -> str:
+        """convert compiled ast to gene_reaction_rule str
+
+        Parameters
+        ----------
+        self : GPR
+           compiled ast Module describing GPR
+        names: dict
+           dictionary of gene ids to gene names. If this is empty, returns gene ids
+
+        Returns
+        ------
+        string
+            The gene reaction rule
+        """
+        return self.__ast2str(self, names=names)
+
+    def __str__(self):
+        """convert compiled ast to gene_reaction_rule str
+
+        Parameters
+        ----------
+        self : GPR
+            compiled ast Module describing GPR
+
+        Returns
+        ------
+        string
+            The gene reaction rule
+        """
+        return self.to_string(names={})
+
+    def __repr_html__(self):
+        return """
+        <table>
```

Comment:
Is there a reason you put it in a table? I think a `<p></p>` would be sufficient.